File: same_solution_with_different_language/Square_of_sorted_array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function (nums) {
    if (!Array.isArray(nums)) {
        throw new TypeError('nums must be a valid Array');
    }
    else if (nums.length === 0) {
        return null;
    }

    let ans = [];

    for (let i = 0; i < nums.length; i++) {
        ans.push(nums[i] ** 2);

    }

    return ans.sort((a, b) => a - b);

};

console.log(sortedSquares([-4, -1, 0, 3, 10]))
console.log(sortedSquares([-7, -3, 2, 3, 11]))

File: Square_of_sorted_array.py
from typing import List


class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        if not isinstance(nums, list):
            raise TypeError(
                'nums must be a valid list object that must contains int object type elements')

        if len(nums) == 0:
            return 0

        nums.sort()
        ans = []

        for i in nums:
            ans.append(i**2)

        ans.sort()
        return ans

  
obj = Solution()
print(obj.sortedSquares([-4, -1, 0, 3, 10]))
print(obj.sortedSquares([-7, -3, 2, 3, 11]))
